archive/a/algol68/sleep-sort.alg

```diff
@@ -0,0 +1,228 @@
+MODE PARSEINT_RESULT = STRUCT(BOOL valid, INT value, STRING leftover);
+MODE PARSEINTLIST_RESULT = STRUCT(BOOL valid, REF []INT values);
+
+PROC parse int = (REF STRING s) PARSEINT_RESULT:
+(
+    BOOL valid := FALSE;
+    REAL r := 0.0;
+    INT n := 0;
+    STRING leftover;
+
+    # Associate string with a file #
+    FILE f;
+    associate(f, s);
+
+    # On end of input, exit if valid number not seen. Otherwise ignore it #
+    on logical file end(f, (REF FILE dummy) BOOL:
+        (
+            IF NOT valid THEN done FI;
+            TRUE
+        )
+    );
+
+    # Exit if value error #
+    on value error(f, (REF FILE dummy) BOOL: done);
+
+    # Convert string to real number #
+    get(f, r);
+
+    # If real number is in range of an integer, convert to integer. Indicate integer is valid if same as real #
+    IF ABS r <= max int
+    THEN
+        n := ENTIER(r);
+        valid := (n = r)
+    FI;
+
+    # Get leftover string #
+    get(f, leftover);
+
+done:
+    close(f);
+    PARSEINT_RESULT(valid, n, leftover)
+);
+
+PROC count list items = (STRING s) INT:
+(
+    INT count := 1;
+    FOR k TO UPB s
+    DO
+        IF s[k] = ","
+        THEN
+            count +:= 1
+        FI
+    OD;
+
+    count
+);
+
+PROC parse int list = (REF STRING s) PARSEINTLIST_RESULT:
+(
+    BOOL valid := FALSE;
+    STRING leftover := s;
+    INT num list items = count list items(s);
+    HEAP [num list items]INT values;
+
+    # Repeat while valid value #
+    FOR k TO num list items
+    DO
+        # Get next integer value and update leftover string #
+        PARSEINT_RESULT result = parse int(leftover);
+        valid := valid OF result;
+        leftover := leftover OF result;
+
+        # Append the integer value to list #
+        values[k] := value OF result;
+
+        # Do nothing if end of string #
+        IF leftover = ""
+        THEN
+            SKIP
+        # Skip comma if leftover string starts with comma #
+        ELIF leftover[1] = ","
+        THEN
+            leftover := leftover[2:]
+        # Otherwise indicate invalid #
+        ELSE
+            valid := FALSE
+        FI
+    UNTIL NOT valid
+    OD;
+
+    PARSEINTLIST_RESULT(valid, values)
+);
+
+PROC usage = VOID: (
+    printf(($gl$, "Usage: please provide a list of at least two integers to sort in the format ""1, 2, 3, 4, 5"""))
+);
+
+COMMENT
+Algol68 does not have any type of mechanism to sleep for a specified time,
```

Comment:
Great comment! This helps me get a feel for which languages are capable of which programs (and which programs we should keep or remove). We have removed programs in the past due to lack of support in most languages.